src/yoroi-wallets/cardano/transactionManager.test.ts

```diff
@@ -0,0 +1,148 @@
+import {fromPairs} from 'lodash'
+import DeviceInfo from 'react-native-device-info'
+
+import {storage as rootStorage, YoroiStorage} from '../storage'
+import {Transaction} from '../types'
+import {mockApi, mockedAddressesByChunks, mockedBackendConfig, mockedHistoryResponse, mockTx} from './mocks'
+import {perAddressTxsSelector, toCachedTx} from './shelley'
+import {makeMemosManager, makeTransactionManager} from './transactionManager'
+
+jest.mock('./api', () => mockApi)
+
+describe('transaction manager', () => {
+  DeviceInfo.getVersion = () => '9.9.9'
+
+  let txManager
+  beforeEach(async () => {
+    txManager = await makeTransactionManager(mockStorage, mockedBackendConfig)
+  })
+
+  afterEach(rootStorage.clear)
+
+  it('stores memos', async () => {
+    expect(txManager.getTransactions()).toEqual({[mockTx.id]: mockTx})
+
+    await txManager.saveMemo(mockTx.id, 'memo 1')
+
+    expect(txManager.getTransactions()).toEqual({[mockTx.id]: {...mockTx, memo: 'memo 1'}})
+  })
+
+  it('gets updated tx cache values', async () => {
+    expect(txManager.getTransactions()).toEqual({[mockTx.id]: mockTx})
+    expect(txManager.getPerAddressTxs()).toEqual(
+      perAddressTxsSelector({
+        transactions: {[mockTx.id]: mockTx},
+        perAddressSyncMetadata: {},
+        bestBlockNum: null,
+      }),
+    )
+    expect(txManager.getPerRewardAddressCertificates()).toEqual({
+      e0acab7e493ece4c1e6ae627ef9f5f7c9b1063e599e4aa91f87f0d58ae: {
+        '0a8962dde362eef1f840defe6f916fdf9701ad53c7cb5dd4a74ab85df8e9bffc': {
+          submittedAt: '2021-09-13T18:42:10.000Z',
+          epoch: 156,
+          certificates: [
+            {
+              kind: 'StakeDeregistration',
+              rewardAddress: 'e0acab7e493ece4c1e6ae627ef9f5f7c9b1063e599e4aa91f87f0d58ae',
+            },
+          ],
+        },
+      },
+    })
+
+    // tx cache mutation
+    await txManager.doSync(mockedAddressesByChunks)
+
+    expect(txManager.getTransactions()).toEqual({
+      [mockTx.id]: mockTx,
+      ...fromPairs(mockedHistoryResponse.transactions.map((t) => [t.hash, toCachedTx(t)])),
```

Comment:
too much logic. we want to avoid needed tests for our tests

Reply:
copied from` transactionCache.test.ts`

Reply:
it is no longer necessary. using snapshots